apps/framework-docs/src/pages/learn-moose/batch-load-data.mdx

```diff
@@ -0,0 +1,362 @@
+                language=repo["language"] or "Multiple Languages",
+                repo_size_kb=repo["size"],
+                created_at=repo["created_at"],
+                updated_at=repo["updated_at"],
+            )
+        )
+    return data
+
+my_function = StreamingFunction(
+    run=fn
+)
+```
+</Python>
+
+
+### Create the Ingest Script
+
+Create a new file for the ingest script:
+
+<TypeScript>
+```bash filename="Terminal" copy
+mkdir -p app/workflows
+touch app/workflows/ingest_stargazers.ts
+```
+</TypeScript>
+
+<Python>
+```bash filename="Terminal" copy
+mkdir -p app/workflows
+touch app/workflows/ingest_stargazers.py
+```
+</Python>
+
+Add the following code to your new file:
+
+<TypeScript>
+```ts filename="app/workflows/ingest_stargazers.ts" copy
+// Coming soon! For now, please use the Python version
+```
+</TypeScript>
+
+<Python>
+```python filename="app/workflows/ingest_stargazers.py" copy
+import os
+import requests
+from datetime import datetime, timedelta
+from dotenv import load_dotenv
+from pathlib import Path
+from typing import Optional, Dict, List
+
+class StargazerIngester:
+    def __init__(self, owner: str, repo: str, token: str, base_url: str):
+        self.owner = owner
+        self.repo = repo
+        self.token = token
+        self.base_url = base_url
+        self.headers = {
+            'Accept': 'application/vnd.github.v3.star+json',
+            'Authorization': f'token {token}'
+        }
+
+    def fetch_stargazers(self, before_date: Optional[datetime] = None) -> List[Dict]:
+        """Fetch all stargazers before a given date"""
+        stargazers = []
+        page = 1
+        per_page = 100
+
+        while True:
+            url = f'https://api.github.com/repos/{self.owner}/{self.repo}/stargazers'
+            params = {'per_page': per_page, 'page': page}
+            
+            response = requests.get(url, headers=self.headers, params=params)
+            if response.status_code != 200:
+                print(f'Error fetching stargazers: {response.status_code} {response.reason}')
+                break
+
+            data = response.json()
+            if not data:
+                break
+
+            # Filter by date if specified
+            if before_date:
+                data = [
+                    star for star in data 
+                    if datetime.strptime(star["starred_at"], "%Y-%m-%dT%H:%M:%SZ") < before_date
+                ]
+                if not data:  # Stop if we've passed our date threshold
+                    break
+
+            stargazers.extend(data)
+            page += 1
+
+        return stargazers
+
+    def ingest_stargazers(self, stargazers: List[Dict]) -> int:
+        """Ingest stargazers into the HistoricalStargazer table via the API endpoint"""
+        ingested_count = 0
+        
+        for stargazer in stargazers:
+            response = requests.post(
+                f"{self.base_url}/ingest/HistoricalStargazer",
+                json={
```

Comment:
This seem to be loading one by one no? not as a batch?
We built the feature where you can send an array of json as the input ie a batch